src/ast/ast_build_filter_tree.c

```diff
@@ -187,10 +244,15 @@ void _AST_ConvertFilters(RecordMap *record_map, const AST *ast,
 	} else if(type == CYPHER_AST_BINARY_OPERATOR) {
 		node = _convertBinaryOperator(record_map, entity);
 	} else if(type == CYPHER_AST_UNARY_OPERATOR) {
-		// TODO, also n-ary maybe
-		assert(false && "unary filters are not currently supported.");
+		node = _convertUnaryOperator(record_map, entity);
 	} else if(type == CYPHER_AST_APPLY_OPERATOR) {
 		assert(false && "APPLY operators are not currently supported in filters.");
+	} else if(type == CYPHER_AST_TRUE) {
+		node = _convertTrueOperator();
+	} else if(type == CYPHER_AST_FALSE) {
+		node = _convertFalseOperator();
+	} else if(type == CYPHER_AST_INTEGER) {
```

Comment:
I think that all the non-pathological entity types are handled here. Given a query like:
`MATCH (a) WHERE 3.4 RETURN a`
We just don't create a filter tree. Neo raises a syntax error: `Type mismatch: expected Boolean but was Float`

An odd consequence of the fairly-similar logic between `_AST_ConvertFilters` and `_FilterNode_FromAST` is that the query:
`MATCH (a) WHERE a.age > 30.4 AND 3.4 RETURN a`
causes the assertion failure `src/ast/ast_build_filter_tree.c:225: _FilterNode_FromAST: Assertion 'false' failed.`
But this doesn't need to be handled here, we should just improve our validations soon.

Reply:
Agree.